src/UartSerial.cpp

```diff
@@ -27,6 +23,8 @@ UartSerial::~UartSerial()
 
 size_t UartSerial::available()
 {
+  if (_fd == -1) return 0;
+
   int numBytesAvailable;
   ioctl(_fd, FIONREAD, &numBytesAvailable);
   return numBytesAvailable;
```

Comment:
Fixed the exit() calls and just left the perror() calls. Added some file descriptor cleanup as well.